### PR TITLE
Add io.github.lavenderdotpet.LibreQuake

### PR DESCRIPTION
**Pull request by pnahratow**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly.
- 
  The LibreQuake project aims to create a complete, free content first-person shooter game. The media data (which this project is primarily about) is provided under the BSD-3-Clause license, To have a playable game this flatpak is bundled with the GPL-2.0 quakespasm sourceport.

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. 
 
I am part of the development team and have implemented this flatpak release on behalf of the owner of the github repo. Will verification be possible afterwards if they login to flathub or do they have to create the submission PR themselve?

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by lavenderdotpet** [comment]
hiii im the lead dev of the project and i approve of this flatpak release

File: io.github.lavenderdotpet.LibreQuake.desktop
Comment (bbhtt) [inline io.github.lavenderdotpet.LibreQuake.desktop:1]: Desktop file should be upstream

File: io.github.lavenderdotpet.LibreQuake.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      <caption>LibreQuake</caption>
+      <image type="source">https://librequake.queer.sh/media/flatpak/storepage05.png</image>
+    </screenshot>
+    <screenshot>
+      <caption>LibreQuake</caption>
+      <image type="source">https://librequake.queer.sh/media/flatpak/storepage01.png</image>
+    </screenshot>
+  </screenshots>
+  <!-- https://hughsie.github.io/oars/generate.html -->
+  <content_rating type="oars-1.0">
```
Comment (bbhtt) [inline io.github.lavenderdotpet.LibreQuake.metainfo.xml:56]:
```suggestion
  <content_rating type="oars-1.1">
```

File: io.github.lavenderdotpet.LibreQuake.metainfo.xml
Comment (bbhtt) [inline io.github.lavenderdotpet.LibreQuake.metainfo.xml:1]: metainfo file should be upstream

File: io.github.lavenderdotpet.LibreQuake.svg
Comment (bbhtt) [inline io.github.lavenderdotpet.LibreQuake.svg:1]: icon should be upstream

File: io.github.lavenderdotpet.LibreQuake.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  - shared-modules/libmad/libmad.json
+
+  - name: quakespasm
+    sources:
+      - type: git
+        url: https://github.com/sezero/quakespasm.git
+        tag: quakespasm-0.96.3
+    buildsystem: simple
+    build-commands:
+      - make --directory=Quake USE_SDL2=1 LDFLAGS="-L/app/lib"
```
Comment (bbhtt) [inline io.github.lavenderdotpet.LibreQuake.yml:25]: Use `make-args`, drop `buildsystem: simple`, move the two install commands to `post-install`

File: io.github.lavenderdotpet.LibreQuake.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    build-commands:
+      - install -d /app/share/librequake
+      - mv id1 /app/share/librequake/
+
+  - name: launch-script
+    sources:
+      - type: script
+        dest-filename: librequake.sh
+        commands:
+          - cp -avu /app/share/librequake/id1 "$XDG_DATA_HOME"/
```
Comment (bbhtt) [inline io.github.lavenderdotpet.LibreQuake.yml:44]: So this will get executed every time it's run.

Why is this needed. This seems weird.

File: io.github.lavenderdotpet.LibreQuake.desktop
Comment (pnahratow) [reply io.github.lavenderdotpet.LibreQuake.desktop:1]: Thank you very much for the review. I don't quite understand. Is this talking about that these files are supposed to live in the original repository instead of this pull request? If yes I will completely remove them here and reference them via file/url in the yaml file. correct?

File: io.github.lavenderdotpet.LibreQuake.yml (hunk 3 above)
Comment (pnahratow) [reply io.github.lavenderdotpet.LibreQuake.yml]: I have found a cleaner solution for this. This was here to persist userdata (config, screenshots etc.). Quakespasm kind of supports user directories natively so I used this now.

File: io.github.lavenderdotpet.LibreQuake.yml (hunk 2 above)
Comment (pnahratow) [reply io.github.lavenderdotpet.LibreQuake.yml]: If I do this. Assuming I understand this correctly. I get 
```
Error: module quakespasm: Can't find autogen, autogen.sh or bootstrap
```
I assume by default the autotools build system is used. This is just a simple makefile project though. The other buildsystems don't seem to fit either. What do?

Comment (bbhtt) [reply io.github.lavenderdotpet.LibreQuake.yml]: Set `no-autogen: true`

Comment (pnahratow) [reply io.github.lavenderdotpet.LibreQuake.yml]: I got it to work.

**Comment by pnahratow** [comment]
I have implemented the changes and ran linter and buildtest again. All good.

**Comment by bbhtt** [comment]
bot, build io.github.lavenderdotpet.LibreQuake
